lib/Config/LocationAwareConfigRepository.php

```diff
@@ -276,8 +277,20 @@ private function invalidateConfigCache(): void
     {
         // invalidate container config cache if debug flag on kernel is set
         $servicesConfig = PIMCORE_PROJECT_ROOT . '/config/services.yaml';
-        if (is_file($servicesConfig)) {
+        if (is_writable($servicesConfig)) {
             touch($servicesConfig);
+        } else {
+            $kernel = Pimcore::getKernel();
+            $app = new Application($kernel);
+            if ($kernel->isDebug()) {
```

Comment:
Do only when necessary:

```suggestion
            if ($kernel->isDebug()) {
                $app = new Application($kernel);
```